solr/solrj-streaming/src/java/org/apache/solr/client/solrj/io/SolrClientCache.java

```diff
@@ -55,6 +50,8 @@ public class SolrClientCache implements Closeable {
   private static final int minSocketTimeout =
       Math.max(Integer.getInteger(HttpClientUtil.PROP_SO_TIMEOUT, MIN_TIMEOUT), MIN_TIMEOUT);
 
+  private String basicAuthCredentials = null; // Only support with the http2SolrClient
```

Comment:
@epugh  I don't agree with this because it's basically a one-off setting when there are many other potential settings.  I don't think basic auth is so special or so extremely widely used that it deserves its very own setter.  It sets an unfortunate precedent from someone interesting in setting something else.  The SolrClientCache _already_ provides a constructor that accepts an existing HTTP client , and that which can be initialized with basic auth.

Reply:
I could get behind that.  It's not always obvious which settings are done on the http client level and which are at the solr level ;-).

Reply:
>  It's not always obvious which settings are done on the http client level and which are at the solr level ;-).

Hmm; I'm not sure how that's relevant to this situation.  If you create an Http2SolrClient (with basic auth and whatever else) and pass it to SolrClientCache, it'll be used completely not partially.

I'm marking this setter deprecated in an upcoming PR loosely related to at least signal we don't want this.